WIP Borg Weapons

NervalARS opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request

El arma taser de los Cyborgs de seguridad a sido remplazada por un arma hibrida que permite tanto disparar un taser como un rayo laser.

## Why It's Good For The Game

Esto dara mas versatilidad a los robots de sec en situación de combate, se acabara el ser forzado a usar el cuerpo a cuerpo para eliminar a un enemigo.

## Changelog
:cl:
add: Added new things
balance: rebalanced something
del: old things
/:cl:

NervalARS committed:
WIP Borg Weapons

NervalARS committed:
WIP 2 Borg Weapons

NervalARS committed:
WIP 3 Borg Weapons

NervalARS committed:
Revert "WIP 3 Borg Weapons"

This reverts commit 1c23ad5ed13d4f466525d808125713db3909b227.

NervalARS committed:
Revert "Revert "WIP 3 Borg Weapons""

This reverts commit 4ffbc2a3f34730e9a509682766075319bdd11f8e.

NervalARS committed:
Revert "Revert "Revert "WIP 3 Borg Weapons"""

This reverts commit 9620b5b928d74cff70aa3ab98b5e6261c4c2d9a3.

NervalARS committed:
WIP 3 Borg Weapons

Silverexperience commented on code/hispania/game/objects/items/weapons/guns/energy/egun.dm:
```diff
@@ -0,0 +1,20 @@
+		list(mode_name="stun", projectile_type=/obj/item/projectile/beam/stun, modifystate="energystun", item_modifystate="stun", fire_sound='sound/weapons/Taser.ogg', icon="stun"),
+		list(mode_name="kill", projectile_type=/obj/item/projectile/beam, modifystate="energykill", item_modifystate="kill", fire_sound='sound/weapons/Laser.ogg', icon="kill"),
+		WEAPON_CHARGE,
+		)
```
Pon un espacio al final del código.

Silverexperience commented on code/modules/projectiles/guns/energy/egun.dm:
```diff
@@ -56,4 +56,5 @@
-		update_mode()
\ No newline at end of file
+		update_mode()
+
```
Quita los dos espacios al final de este archivo

Silverexperience commented on code/modules/projectiles/guns/energy/egun.dm:
```diff
@@ -56,4 +56,4 @@
 	if(cell && cell.charge >= charge_cost) //no overlay if we dont have any power
-		update_mode()
\ No newline at end of file
+		update_mode()
```
Te dejaste un espacio al final del código.

Silverexperience commented on code/modules/mob/living/silicon/robot/robot_upgrades.dm:
```diff
@@ -128,7 +128,7 @@
 		return 0
 
-	var/obj/item/weapon/gun/energy/taser/mounted/cyborg/T = locate() in R.module
+	var/obj/item/weapon/gun/energy/gun/mounted/cyborg/T = locate() in R.module
```
¿Por qué cambias la mejora del taser al del arma hibrida?
Y si lo quieres cambiar cambia también el texto que no tiene sentido que al poner la mejora te digan que has añadido un taser pero en realidad es un arma hibrida.

Silverexperience commented on code/hispania/game/objects/items/weapons/guns/energy/egun.dm:
```diff
@@ -0,0 +1,20 @@
+/obj/item/weapon/gun/energy/gun/mounted/cyborg
```
La mayoría de las cosas de este objeto son heredadas; podrías quitarlas.

Silverexperience reviewed (changes requested):
- Cuando disparas el arma tu puntero cambia y solo se puede quitar cambiando de modo o aveces ni se quita.

- Decias que con alt + click se puede cambiar el seguro, pero no parece funcionar.

NervalARS commented:
El seguro fue retirado, una vez que se cambió el arma a la versión montada.
En respecto al puntero lo corregiré.

El jue., 24 sept. 2020 13:39, Silverexperience <notifications@github.com>
escribió:

> *@Silverexperience* requested changes on this pull request.
>
>    -
>
>    Cuando disparas el arma tu puntero cambia y solo se puede quitar
>    cambiando de modo o aveces ni se quita.
>    -
>
>    Decias que con alt + click se puede cambiar el seguro, pero no parece
>    funcionar.
>
> ------------------------------
>
> In code/modules/projectiles/guns/energy/egun.dm
> <https://github.com/Helixis/CEV-Eris/pull/32#discussion_r494233077>:
>
> > \ No newline at end of file
>
> +		update_mode()
>
>
> Te dejaste un espacio al final del código.
> ------------------------------
>
> In code/modules/mob/living/silicon/robot/robot_upgrades.dm
> <https://github.com/Helixis/CEV-Eris/pull/32#discussion_r494235775>:
>
> > @@ -128,7 +128,7 @@
>
>  		to_chat(usr, "There's no mounting point for the module!")
>
>  		return 0
>
>
>
> -	var/obj/item/weapon/gun/energy/taser/mounted/cyborg/T = locate() in R.module
>
> +	var/obj/item/weapon/gun/energy/gun/mounted/cyborg/T = locate() in R.module
>
>
> ¿Por qué cambias la mejora del taser al del arma hibrida?
> Y si lo quieres cambiar cambia también el texto que no tiene sentido que
> al poner la mejora te digan que has añadido un taser pero en realidad es un
> arma hibrida.
> ------------------------------
>
> In code/hispania/game/objects/items/weapons/guns/energy/egun.dm
> <https://github.com/Helixis/CEV-Eris/pull/32#discussion_r494243309>:
>
> > @@ -0,0 +1,20 @@
>
> +/obj/item/weapon/gun/energy/gun/mounted/cyborg
>
>
> La mayoría de las cosas de este objeto son heredadas; podrías quitarlas.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/Helixis/CEV-Eris/pull/32#pullrequestreview-495464817>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ANMQR5FEA3UJGQKB2JVXTSDSHMVV3ANCNFSM4RWJY7ZA>
> .
>

NervalARS committed:
WIP 4 Borg Weapons

Se ha corregido el problema con la bateria.

El problema de la mira no parece tener solucion.

NervalARS committed:
WIP 5 Borg Weapons

Correción de codigo menor.

NervalARS committed:
WIP 6 Borg Weapons

Correción de texto.

Evankhell561 commented on code/hispania/game/objects/items/weapons/guns/energy/egun.dm:
```diff
@@ -0,0 +1,33 @@
+/obj/item/weapon/gun/energy/egun
```
Está ruta no existía ya? Parece que la estás duplicado.

Evankhell561 reviewed (changes requested):
El pr estaba casi listo, silver sólo te pidió unas pequeñas cosas pero parece que te has vuelto un lío. Recomendaría dar uno o dos pasos atrás.

NervalARS committed:
WIP 7 Borg Weapons

Las rutas se han simplificado.

NervalARS committed:
WIP 8 Borg Weapons

Se ha simplificado las rutas.

El modulo emag del borg de seguridad ha sido cambiado a un cañon de plasma, para que no sea redudante.

Silverexperience reviewed (changes requested):
- El puntero sigue teniendo el problema de no poder quitarse.

- Ahora la pistola ya no recoge la energia del borg.

- Sigue estando la linea del final en  code/modules/projectiles/guns/energy/egun.dm.

Silverexperience reviewed (changes requested):
- El puntero sigue igual.

- Cuando decia que tenia cosas heredadas no me referia a que quites las herencias y las pegues ahi, puede quedar asi pero es un poco lioso e inecesario.

NervalARS committed:
WIP 9 Borg Weapons

Se ha eliminado las linea sobrante en code\modules\projectiles\guns\energy\egun.dm.

El arma recoge energia de la bateria del borg para recargarse.

Evankhell561 commented on code/hispania/game/objects/items/weapons/guns/energy/egun.dm:
```diff
@@ -0,0 +1,23 @@
+	icon_state = "energystun100"
+	item_state = null	//so the human update icon uses the icon_state instead.
+	item_charge_meter = TRUE
+	can_dual = 1
```
= TRUE

NervalARS replied:
No entiendo lo que intentas decir. El item_state deberia estar en TRUE?

Evankhell561 commented on code/hispania/game/objects/items/weapons/guns/energy/egun.dm:
```diff
@@ -0,0 +1,23 @@
+		list(mode_name="stun", projectile_type=/obj/item/projectile/beam/stun, modifystate="energystun", item_modifystate="stun", fire_sound='sound/weapons/Taser.ogg', icon="stun"),
+		list(mode_name="kill", projectile_type=/obj/item/projectile/beam, modifystate="energykill", item_modifystate="kill", fire_sound='sound/weapons/Laser.ogg', icon="kill"),
+		WEAPON_CHARGE,
+	)
```
Por una línea nueva acá.

NervalARS committed:
WIP Final Borg Weapons

Se ha optimizado el codigo. Y se ha añadido la linea que faltaba.

Silverexperience commented on code/hispania/game/objects/items/weapons/guns/energy/egun.dm:
```diff
@@ -0,0 +1,8 @@
+/obj/item/weapon/gun/energy/gun/mounted/cyborg
+	name = "Hybrid Gun"
+	desc = "Hybrid Gun is a versatile energy based sidearm, capable of switching between low and high capacity projectile settings. In other words: Stun or Kill."
+	charge_meter = FALSE
```
¿Porque lo pones falso? Poder ver las balas que quedan en el sprite no es algo malo.

Silverexperience commented on code/modules/mob/living/silicon/robot/robot_modules.dm:
```diff
@@ -711,17 +711,17 @@ var/global/list/robot_modules = list(
 	//src.modules += new /obj/item/device/holowarrant(src)
 	src.modules += new /obj/item/weapon/book/manual/wiki/security_ironparagraphs(src) // book of ironhammer paragraphs
-	src.emag = new /obj/item/weapon/gun/energy/laser/mounted(src)
+	src.emag = new /obj/item/weapon/gun/energy/plasma/mounted/blitz(src)
```
¿Porque este cambio de armas en el emag?

NervalARS replied:
Para que el borg no tenga dos armas laser

Silverexperience commented on code/hispania/game/objects/items/weapons/guns/energy/egun.dm:
```diff
@@ -0,0 +1,8 @@
+	charge_meter = FALSE
+	zoom_factor = 0
+	charge_cost = 100
+	recharge_time = 10
```
Podrias reducir el tiempo de carga. Tarda bastante y al usar energia del borg tampoco tendras infinitos disparos.

NervalARS committed:
WIP Borg Weapon

Se redujo el tiempo de recarga del arma de los borg.

NervalARS committed:
WIP Borg Weapon

charge_meter ahora es TRUE
